src/mcp/http-adapter.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* c8 ignore start */
+import { Buffer } from 'buffer';
+
+export function toNodeRequest(fetchRequest) {
+  const headers = {};
+  fetchRequest.headers.forEach((value, key) => {
+    headers[key.toLowerCase()] = value;
+  });
+  return {
+    method: fetchRequest.method,
+    headers,
+    // minimal event emitter interface expected by transport
+    on() {},
+  };
+}
+
+export function createMockResponse() {
+  const chunks = [];
+  let totalLength = 0;
+  const headers = {};
+  let statusCode = 200;
+  let resolveDone;
+  const done = new Promise((resolve) => {
+    resolveDone = resolve;
+  });
+
+  const MAX_BUFFER_SIZE = 4 * 1024 * 1024; // 4 MB
```

Comment:
Is this the same as `MAX_BODY_SIZE`? I see this math done 2 times across the PR. If they are the same, I'd probably define a central constant for each